File: flutter/lib/domain/post/usecases/reactToPostUseCase.dart
import '../postRepostiory.dart';

import '../post.dart';
import '../reaction.dart';

class ReactToPostUseCase {
  PostRepository postRepository;

  ReactToPostUseCase(PostRepository postRepository) {
    this.postRepository = postRepository;
  }

  Future<Post> run(Reaction reaction) {
    return postRepository.reactToPost(reaction);
  }
}

File: web/src/domain/user/usecases/ReactToPostUseCase.ts
import Post from "../../post/Post";
import PostRepository from "../../post/PostRepository";
import Reaction from "../../post/Reaction";

export default class ReactToPostUseCase {
    postRepository: PostRepository;

    constructor(postRepository: PostRepository) {
        this.postRepository = postRepository;
    }

    async run(reaction: Reaction): Promise<Post> {
        return await this.postRepository.reactToPost(reaction);
    }
}